cmd/promtool/analyze_test.go

```diff
@@ -136,7 +137,9 @@ func TestGetBucketCountsAtTime(t *testing.T) {
 	}
 
 	for _, c := range cases {
+		c := c
```

Comment:
After Go 1.22 [you don't need to re-assign variable](https://go.dev/blog/loopvar-preview). It's also applied for other tests

Reply:
I did not know about this, thanks. 
Modified my PR to avoid unnecessary closure.